code/modules/shuttle/on_move.dm

```diff
@@ -393,13 +393,34 @@ All ShuttleMove procs go here
 /obj/docking_port/stationary/onShuttleMove(turf/newT, turf/oldT, rotation, list/movement_force, move_dir, old_dock)
 	if(old_dock == src) //Don't move the dock we're leaving
 		return FALSE
+	. = ..()
 
+/obj/docking_port/mobile/fob/beforeShuttleMove(turf/newT, rotation, move_mode)
+	src.previous_dock = src.get_docked()
 	. = ..()
 
+/obj/docking_port/mobile/fob/afterShuttleMove(list/movement_force, shuttle_dir, shuttle_preferred_direction, move_dir)
+	. = ..()
+	//Keep the planet loaded
+	var/obj/docking_port/mobile/fob/M = src
+	var/obj/docking_port/stationary/fob_land/LZ = M.get_docked()
```

Comment:
Stop using all caps names if they're longer than 1 char and not a define. Use something descriptive, that was half the point of the dock rewrite.